### PR TITLE
Elavon: Update Stored Credentials

### PR DESCRIPTION
**Pull request by almalee24**
To satisfy new Elavon API requirements, including recurring_flag, approval_code for non-tokenized PMs, installment_count and _number, and situational par_value and association_token_data fields.

Remote
40 tests, 178 assertions, 2 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 95% passed

Unit
60 tests, 356 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

### CODE REVIEW

File: lib/active_merchant/billing/gateways/elavon.rb
Comment (DustinHaefele) [inline lib/active_merchant/billing/gateways/elavon.rb:404]: These are new fields that we don't send today right? Will they for sure be present when we are trying to add them? 

I guess my overall question is should adding them here be conditional on whether they are present like 

```suggestion
        xml.ssl_payment_count options[:installments] if options[:installments]
```

Comment (almalee24) [reply lib/active_merchant/billing/gateways/elavon.rb:404]: We won't merge these changes until these fields are being passed for Installment transactions are being performed. They will start to fail since these are required fields.

File: lib/active_merchant/billing/gateways/elavon.rb
Comment (almalee24) [reply lib/active_merchant/billing/gateways/elavon.rb:199]: Will if they are passing `ssl_token` the that's the only thing they will need for subsequent transactions. You can find this under the Elavon Stored Credentials section in the documeantion. It's a bit confusing but in the initial transaction customer will get back ssl_token and they can use this on subsequent transactions. ssl_token is the tokenized payment method

File: lib/active_merchant/billing/gateways/elavon.rb
Comment (DustinHaefele) [reply lib/active_merchant/billing/gateways/elavon.rb:404]: So will these fields need to be passed for all customers before merging?

Comment (almalee24) [reply lib/active_merchant/billing/gateways/elavon.rb:404]: Just customers who pass `reason_type: 'installment'` which is only 2 and we've communicated with them. I'm waiting for confirmation and I'll check on our side and then I'll merge.

File: lib/active_merchant/billing/gateways/elavon.rb
Comment (DustinHaefele) [inline lib/active_merchant/billing/gateways/elavon.rb:375]: In the ticket it mentions the sending the `ssl_association_token`, but we are sending `ssl_association_token_data` Is there any issue with that?

**Review by DustinHaefele — COMMENTED** [review]
Alright I am following this now. I feel pretty good about it. Left one more question and then I'll 👍

File: lib/active_merchant/billing/gateways/elavon.rb
Comment (almalee24) [reply lib/active_merchant/billing/gateways/elavon.rb:375]: I'm not entirely sure to be honest. I basically copied and pasted Davids past PR that he used for certification. Let me look at the docs and I'll get back to you

Comment (almalee24) [reply lib/active_merchant/billing/gateways/elavon.rb:375]: Looking at their docs I'm thinking it was a type because only ssl_association_token_data comes up

**Review by DustinHaefele — APPROVED** [review]
This looks good. Thanks for answering all of my questions!

**Review by DustinHaefele — APPROVED** [review]

**Pull request merged**
